src/github_integration/github_manager.py

```diff
@@ -0,0 +1,231 @@
+            self.__git_release = release
+
+        except Exception as e:
+            if "Not Found" in str(e):
+                logging.error(f"Latest release not found for {self.__repository.full_name}. 1st release for repository!")
+                self.__git_release = None
+            else:
+                logging.error(f"Fetching latest release failed for {self.__repository.full_name}: {str(e)}. "
+                              f"Expected first release for repository.")
+                self.__git_release = None
+
+        return self.__git_release
+
+    def fetch_issues(self, since: datetime = None, state: str = None) -> list[Issue]:
+        """
+        Fetches all issues from the current repository.
+        If a since is set, fetches all issues since the defined time.
+
+        :return: A list of Issue objects.
+        """
+        if self.__git_release is None:
+            logging.info(f"Fetching all issues for {self.__repository.full_name}")
+            issues = self.__repository.get_issues(state="all")
+        else:
+            logging.info(f"Fetching all issues since {self.__git_release.published_at} for {self.__repository.full_name}")
+            issues = self.__repository.get_issues(state="all", since=self.__git_release.published_at)
+
+        parsed_issues = []
+        logging.info(f"Found {len(list(issues))} issues for {self.__repository.full_name}")
+        for issue in list(issues):
+            parsed_issues.append(Issue(issue))
+
+        return parsed_issues
+
+    def fetch_pull_requests(self, since: datetime = None, state: str = None) -> list[PullRequest]:
+        """
+        Fetches all pull requests from the current repository.
+        If a 'since' datetime is provided, fetches all pull requests since that time.
+        If a 'state' is provided, fetches pull requests with that state.
+
+        :param since: The datetime to fetch pull requests since. If None, fetches all pull requests.
+        :param state: The state of the pull requests to fetch. If None, fetches pull requests of all states.
+        :return: A list of PullRequest objects.
+        """
+        logging.info(f"Fetching all closed PRs for {self.__repository.full_name}")
+        if state is None:
+            pulls = self.__repository.get_pulls()
+        else:
+            pulls = self.__repository.get_pulls(state=state)
+
+        pull_requests = []
+        logging.info(f"Found {len(list(pulls))} PRs for {self.__repository.full_name}")
+        for pull in list(pulls):
+            pull_requests.append(PullRequest(pull))
+
+        return pull_requests
+
+    def fetch_commits(self, since: datetime = None) -> list[Commit]:
+        logging.info(f"Fetching all commits {self.__repository.full_name}")
+        raw_commits = self.__repository.get_commits()
+
+        commits = []
+        for raw_commit in raw_commits:
+            # for reference commit author - use raw_commit.author
```

Comment:
som dead code below

Reply:
I am working on this class now. I am adding a unit test. I am doing a cleanup during this action.